cpp/tensorrt_llm/batch_manager/cacheTransceiver.cpp

```diff
@@ -58,6 +63,72 @@ namespace tensorrt_llm::batch_manager
 
 std::mutex CacheTransceiver::mDllMutex;
 
+TransferTagClient::TransferTagClient()
+{
+    mContext = std::make_unique<zmq::context_t>(1);
+}
+
+uint64_t TransferTagClient::getTransferTag(std::string const& serverEndpoint, RequestIdType const& receiverTransferId,
+    UuidType const& receiverServerUuid, int32_t expectedRefCount)
+{
+    std::lock_guard<std::mutex> lock(mMutex);
+
+    zmq::socket_t socket(*mContext, zmq::socket_type::dealer);
+    socket.connect(serverEndpoint);
```

Comment:
From what I can read online, creating a new socket for each request is costly. Could we create a cache?

```
private:
    std::unordered_map<std::string, std::unique_ptr<zmq::socket_t>> mSocketCache;
    
zmq::socket_t& getOrCreateSocket(std::string const& serverEndpoint) {
    auto it = mSocketCache.find(serverEndpoint);
    if (it == mSocketCache.end()) {
        auto socket = std::make_unique<zmq::socket_t>(*mContext, zmq::socket_type::dealer);
        socket->connect(serverEndpoint);
        it = mSocketCache.emplace(serverEndpoint, std::move(socket)).first;
    }
    return *it->second;
}
```

might be worth also talking to others on team who have more experience with ZeroMQ.